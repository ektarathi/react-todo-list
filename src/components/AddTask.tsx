import * as React from 'react';
import { makeStyles } from "@material-ui/core/styles";
import TextField from "@material-ui/core/TextField";
import { useDispatch } from "react-redux";
import { TaskActions } from "../redux/actions/taskAction";
import Button from "@material-ui/core/Button";
import useLocalStorage from './useLocalStorege';

const useStyles = makeStyles(theme => ({
    root: {
      "& > *": {
        margin: theme.spacing(1),
        width: 200
      }
    },
    addTask: {
      marginTop: 15
    }
}));

export interface AddTaskProps {
    
}
 
const AddTask: React.SFC<AddTaskProps> = () => {
    const classes = useStyles();
    const [value, setValue] = useLocalStorage('item', '');
    const nameDispatch = useDispatch<React.Dispatch<TaskActions>>();

  const handleSubmit = (e:any) => {
    e.preventDefault();
    nameDispatch({ type: "ADD_TASK", payload: value});
    setValue("");
  };

  return (
    <form className={classes.root}>
      <TextField
        type="text"
        className="input"
        label="Enter task name"
        value={value}
        onChange={e => setValue(e.target.value)}
      />
      <Button
        variant="contained"
        color="primary"
        onClick={handleSubmit}
        className={classes.addTask}
      >
        Add Task
      </Button>
    </form>
  );
};
 
export default AddTask;