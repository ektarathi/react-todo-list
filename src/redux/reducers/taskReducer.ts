import { TaskActions } from "../actions/taskAction";

const NameReducer = (taskList = [] as any, action: TaskActions) => {
	switch (action.type) {
		case "ADD_TASK":
			return [
				...taskList,
				{
					taskName: action.payload,
				},
			];
		case "COMPLETE_TASK":
			console.log("welcome:", taskList, taskList[action.taskIndex]);
			const completedTask = [...taskList];
			completedTask[action.taskIndex].isCompleted = true;
			return completedTask;
		case "DELETE_TASK":
			const deletedTask = [...taskList];
			deletedTask.splice(action.taskIndex, 1);
			return deletedTask;
		default:
			return taskList;
	}
};

export default NameReducer;
