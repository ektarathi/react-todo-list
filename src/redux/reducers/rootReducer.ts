import { combineReducers } from 'redux';
import taskReducer from './taskReducer';

const rootReducer = combineReducers({
    task: taskReducer,
})

export type AppState = ReturnType<typeof rootReducer>
export default rootReducer;