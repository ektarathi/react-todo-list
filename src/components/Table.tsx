import * as React from "react";
import { useSelector } from "react-redux";
import { AppState } from "../redux/reducers/rootReducer";
import Task from "./Task";
import AddTask from "./AddTask";
import { makeStyles } from "@material-ui/core/styles";
import Table from "@material-ui/core/Table";
import TableBody from "@material-ui/core/TableBody";
import TableCell from "@material-ui/core/TableCell";
import TableContainer from "@material-ui/core/TableContainer";
import TableHead from "@material-ui/core/TableHead";
import TableRow from "@material-ui/core/TableRow";
import Paper from "@material-ui/core/Paper";
import Typography from "@material-ui/core/Typography";

const useStyles = makeStyles({
	table: {
		minWidth: 250,
	},
});

export interface TableProps {}

const TableList: React.SFC<TableProps> = () => {
	const classes = useStyles();
    const tasks = useSelector((state: AppState) => state.task);

	return (
		<React.Fragment>
			{tasks.length === 0 ? (
                <Typography variant="h6" component="h2"> Add Task to the list:</Typography>
			) : (
				<TableContainer component={Paper}>
					<Table className={classes.table} aria-label="simple table">
						<TableHead>
							<TableRow>
								<TableCell>Name</TableCell>
								<TableCell>Completed Task </TableCell>
								<TableCell>Deleted Task </TableCell>
							</TableRow>
						</TableHead>
						<TableBody>
							{tasks.map((task: any, index: any) => (
								<Task key={index} index={index} task={task} />
							))}
						</TableBody>
					</Table>
				</TableContainer>
			)}
			<AddTask />
		</React.Fragment>
	);
};

export default TableList;
