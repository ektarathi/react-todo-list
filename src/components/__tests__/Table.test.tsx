import React from 'react';
import { Provider } from 'react-redux';
import { useSelector } from "react-redux";
import { AppState } from "../../redux/reducers/rootReducer";
import configureStore from 'redux-mock-store';
import Enzyme, { shallow, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import AddTask from '../AddTask';
import Table from '../Table';

Enzyme.configure({ adapter: new Adapter() });

describe('test Table Component', () => {
    const expected = [{taskName: "sddds"}];
    const mockStore = configureStore();
    let store;

    it('matches even if received contains additional elements', () => {
        expect([{taskName: "sddds"}]).toEqual(expect.arrayContaining(expected));
    });

    it('does not match if received does not contain expected elements', () => {
        expect([{taskName: "welcome user"}]).not.toEqual(expect.arrayContaining(expected));
    });

    it("Should capture content correctly onChange", () => {
        store = mockStore(0);
        const tasks = useSelector((state: AppState) => state.task);
        const wrapper = mount(<Provider store={store}><AddTask /></Provider>);
        const content = wrapper.find("input").at(0);
        content.value = "MCD";
        content.simulate("change", { target: {value: 'MCD'}});
        
    });
});