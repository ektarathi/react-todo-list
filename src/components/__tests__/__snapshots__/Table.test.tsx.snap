// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Table Component renders Table Component 1`] = `ReactWrapper {}`;
